scripts/start/readCheckScenarioConfig.R

```diff
@@ -43,13 +43,41 @@ readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE
       whitespaceerrors <- whitespaceerrors + sum(haswhitespace)
     }
   }
-  errorsfound <- sum(toolong) + sum(regionname) + sum(illegalchars) + whitespaceerrors
   if ("path_gdx_ref" %in% names(scenConf) && ! "path_gdx_refpolicycost" %in% names(scenConf)) {
     scenConf$path_gdx_refpolicycost <- scenConf$path_gdx_ref
     message("In ", filename,
         ", no column path_gdx_refpolicycost for policy cost comparison found, using path_gdx_ref instead.")
   }
   scenConf[, names(path_gdx_list)[! names(path_gdx_list) %in% names(scenConf)]] <- NA
+
+  # fill empty cells with values from scenario written in copyConfigFrom cell
+  copyFromMissing <- NULL
+  copyFromLater <- NULL
+  if ("copyConfigFrom" %in% names(scenConf)) {
+    copyFromMissing <- setdiff(scenConf[, "copyConfigFrom"], c(NA, rownames(scenConf)))
+    copyFromLater <- ! (match(scenConf[, "copyConfigFrom"], rownames(scenConf)) < seq_along(rownames(scenConf)) | 
+                     is.na(scenConf[, "copyConfigFrom"]) | scenConf[, "copyConfigFrom"] %in% copyFromMissing)
+    if (length(copyFromMissing) > 0) {
+      warning("The following scenario names indicated in copyConfigFrom column were not found in scenario list: ",
+              paste0(copyFromMissing, collapse = ", "), ". Stopping now.")
+    }
+    if (any(copyFromLater)) {
+      warning("The following scenarios references in copyConfigFrom column a scenario defined only later: ",
+              paste0(rownames(scenConf)[copyFromLater], collapse = ", "), ". Stopping now.")
+    }
+    for (run in rownames(scenConf)) {
+      copyConfigFrom <- scenConf[run, "copyConfigFrom"]
+      if (! is.na(copyConfigFrom)) {
+        scenConf[run, is.na(scenConf[run, ])] <- scenConf[copyConfigFrom, is.na(scenConf[run, ])]
+      }
+    }
+    scenConf$copyConfigFrom <- NULL
+  }
+
+  errorsfound <- sum(toolong) + sum(regionname) + sum(illegalchars) + whitespaceerrors +
+                 length(copyFromMissing) + sum(copyFromLater)
+
```

Comment:
Could you put this into a separate function, which takes `scenConf` as an argument and fills it out according to `copyConfigFrom` and returns it? I think that way it would be much easier to understand what is going on, and also it would be possible to write some unit tests for this function, asserting that the filling works as intended.